jdbc-v2/src/main/java/com/clickhouse/jdbc/StatementImpl.java

```diff
@@ -262,10 +267,10 @@ public void cancel() throws SQLException {
             return;
         }
 
-        try {
-            connection.client.query(String.format("KILL QUERY%sWHERE query_id = '%s'",
-                    connection.onCluster ? " ON CLUSTER " + connection.cluster + " " : " ",
-                    lastQueryId), connection.getDefaultQuerySettings()).get();
+        try (QueryResponse response = connection.client.query(String.format("KILL QUERY%sWHERE query_id = '%s'",
+                connection.onCluster ? " ON CLUSTER " + connection.cluster + " " : " ",
+                lastQueryId), connection.getDefaultQuerySettings()).get()){
```

Comment:
`lastQueryId` is set only after request is complete (see com.clickhouse.jdbc.StatementImpl#executeQuery(java.lang.String, com.clickhouse.client.api.query.QuerySettings). It should be generated.

The same for `com.clickhouse.jdbc.StatementImpl#executeUpdate(java.lang.String, com.clickhouse.client.api.query.QuerySettings)`